src/main/resources/data/computercraft/lua/bios.lua

```diff
@@ -758,14 +766,16 @@ function fs.complete(sPath, sLocation, bIncludeFiles, bIncludeDirs)
             if #sFile >= #sName and string.sub(sFile, 1, #sName) == sName then
                 local bIsDir = fs.isDir(fs.combine(sDir, sFile))
                 local sResult = string.sub(sFile, #sName + 1)
-                if bIsDir then
-                    table.insert(tResults, sResult .. "/")
-                    if bIncludeDirs and #sResult > 0 then
-                        table.insert(tResults, sResult)
-                    end
-                else
-                    if bIncludeFiles and #sResult > 0 then
-                        table.insert(tResults, sResult)
+                if bIncludeHidden or string.sub(sFile, 1, 1) ~= "." or string.sub(sResult, 1, 1) ~= "." or sResult ~= sFile then
```

Comment:
Just going through this logic, just wondering what the `sResult ~= sFile` is for? I couldn't find a case where it affected things.

An alternative way of doing this would be to replace the if statement on line 766 with the following:

```lua
if #sFile >= #sName and string.sub(sFile, 1, #sName) == sName and (
    bIncludeHidden or sFile:sub(1, 1) ~= "." or sName:sub(1, 1) == "."
) then
```
It appears to be have the same, but you've definitely looked at this more carefully than me, so will defer to you!

Reply:
I added the comparison for the hidden file with extension case @Wojbie mentioned.
I have to check if that works for completing hidden files after inserting the dot, since sName and sResult are different, no?

Reply:
I looked at it, your change makes sense, pushing it now